src/main/java/com/google/enterprise/sessionmanager/ArtifactStorageService.java

```diff
@@ -0,0 +1,44 @@
+package com.google.enterprise.sessionmanager;
+
+import java.util.Iterator;
+import javax.inject.Inject;
+import org.opensaml.common.binding.artifact.SAMLArtifactMap.SAMLArtifactMapEntry;
+import org.opensaml.util.storage.StorageService;
+
+public class ArtifactStorageService implements StorageService<String, SAMLArtifactMapEntry> {
```

Comment:
can the redis repository implement this interface instead? Or have a method that returns an implementation of this interface? so that the main code is independent of vendor-specific (implementation) code. We might want to support different implementations later.

Is this artifact storage service making ALL saml artifacts to be remotely persisted / loaded back from ?! If so, do we know what effect does this have on performance / network traffic?